### PR TITLE
PoC for a richer navigation

### PR DESCRIPTION
**Pull request by ignasi35**
Introduces a more detailed navigation. Instead of:
Screenshot 2020-09-02 at 18 46 54

There's more detailed info and direct links to specific parts of the guide:
Screenshot 2020-09-02 at 18 46 48

-----------------------------

This PR is a proof-of-concept to start exploring different ways to organize the guide:

1. each component declares it's own `nav.adoc` and `antora.yml` reorders them
1. a richer nav allows reconsidering the contents of `module/pages/index.adoc` since it is less likely users will land there. See for example how `modules/concepts/index.adoc` is no longer a catalogue of links but an actual introduction (meaningful content)
1. IIUC, the information in `nav:` on `antora.yml` is the source for [automatic page navigation](https://github.com/akka/akka-platform-guide/issues/91#issuecomment-685843159)

### CODE REVIEW

File: docs-source/docs/modules/how-to/nav.adoc [hunk 1]
```diff
@@ -0,0 +1 @@
+* xref:index.adoc[]
```
Comment (patriknw) [inline docs-source/docs/modules/how-to/nav.adoc:1]: could you also fix the superfluous `*` in index.adoc
```
== * xref:how-to:index.adoc[]
```

**Review by patriknw — COMMENTED** [review]
looking good, but I guess the concepts structure is to be decided

**Comment by ennru** [comment]
Great, I had just started on a similar urge.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**